package/MDAnalysis/core/universe.py

```diff
@@ -244,113 +306,67 @@ class Universe(object):
         master ConnectivityGroups for each connectivity type
 
     """
+    _topology = None
+
+    
+    def __init__(self, topology, *coordinates, all_coordinates=False,
+                 format=None, topology_format=None, transformations=None,
+                 guess_bonds=False, vdwradii=None, anchor_name=None,
+                 is_anchor=True, in_memory=False, in_memory_step=1,
+                 **kwargs):
 
-    def __init__(self, *args, **kwargs):
-        # Store the segments for the deprecated instant selector feature.
-        # This attribute has to be defined early to avoid recursion in
-        # __getattr__.
-        self._instant_selectors = {}
-        # hold on to copy of kwargs; used by external libraries that
-        # reinitialize universes
-        self._kwargs = copy.deepcopy(kwargs)
-
-        # managed attribute holding Reader
-        self._trajectory = None
+        self._instant_selectors = {}  # for storing segments. Deprecated?
+        self._trajectory = None  # managed attribute holding Reader
         self._cache = {}
+        self._anchor_name = anchor_name
+        self.is_anchor = is_anchor
+        self.atoms = None
+        self.residues = None
+        self.segments = None
+        self.filename = None
+
+        self._kwargs = {
+            'transformations': transformations,
+            'guess_bonds': guess_bonds,
+            'vdwradii': vdwradii,
+            'anchor_name': anchor_name,
+            'is_anchor': is_anchor,
+            'in_memory': in_memory,
+            'in_memory_step': in_memory_step,
+            'format': format,
+            'topology_format': topology_format,
+            'all_coordinates': all_coordinates
+        }
+        self._kwargs.update(kwargs)
+
+        format, topology_format = _resolve_formats(*coordinates, format=format,
+                                                   topology_format=topology_format)
+
+        if not isinstance(topology, Topology) and not topology is None:
+            self.filename = _check_file_like(topology)
+            topology = _topology_from_file_like(self.filename, 
+                                                topology_format=topology_format,
+                                                **kwargs)
+
+        self._topology = topology
+
+        if topology is not None:
+            self._generate_from_topology()  # make real atoms, res, segments
```

Comment:
To keep the ongoing pattern, can remove `_generate_from_topology()` from the class and place it above with your other tidying functions.  Maybe it's cleaner to not have methods in the Universe namespace that can't be called after `__init__`

Reply:
We could -- but _generate_from_topology() modifies the instance, unlike the other methods. Other methods like `_gen_anchor_hash` and `_process_attr` also modify self and shouldn't really be called after `__init__` or by the user. I think it's somewhat common to have code that should only run in `__init__` wrapped in other functions to make code more readable. Moving `_generate_from_topology` outside would result in the below, which looks a bit weird.
```suggestion
            _generate_from_topology(self)  # make real atoms, res, segments
```

By the way, I think you could just move the contents of `_gen_anchor_hash` to `anchor_name`...

```python
    @property
    def anchor_name(self):
        return self._gen_anchor_hash()

    @anchor_name.setter
    def anchor_name(self, name):
        self.remove_anchor()  # clear any old anchor
        self._anchor_name = str(name) if not name is None else name
        self.make_anchor()  # add anchor again

    def _gen_anchor_hash(self):
        # hash used for anchoring.
        # Try and use anchor_name, else use (and store) uuid
        if self._anchor_name is not None:
            return self._anchor_name
        else:
            try:
                return self._anchor_uuid
            except AttributeError:
                # store this so we can later recall it if needed
                self._anchor_uuid = uuid.uuid4()
                return self._anchor_uuid
```

Reply:
Also, correct me if I'm wrong, but I think calling it again after __init__ would have only have a real change on the universe if the topology has changed, which does seem useful in certain situations.

Reply:
Yeah squashing in _gen_anchor_hash seems like a good idea.

I have no idea what calling `_generate_from_topology` would do post init, so probably not a good idea to do so.  It redefines what the `AtomGroup` class fundamentally is, so old AtomGroups would.... stop working?  I don't mind either way on `gen_from_top`.